Front-end/src/Meetup.jsx

```diff
@@ -179,17 +183,73 @@ const Meetup = () => {
                 },
               }
             ).then((location) =>
-              location.json().then((location) => {
-                setPlaceId(location.results[0].place_id);
-                setIsLoading(false);
-              })
+              location
+                .json()
+                .then((location) => {
+                  //Coordinates store every waypoint that a car would be in the route throughout the trip
+                  let coordinates =
+                    location.features[0].geometry.coordinates[0];
+                  return Promise.all([
+                    findCoordinates(
+                      coordinates,
+                      middleLatitude,
+                      middleLongitude
+                    ),
+                  ]);
+                })
+                .then((coordinates) => {
+                  reverseGeocode(coordinates[0][1], coordinates[0][0]);
+                })
             );
+          } else {
+            reverseGeocode(middleLatitude, middleLongitude);
           }
         })
       );
     }
   }
 
+  async function reverseGeocode(latitude, longitude) {
+    fetch(
+      `https://api.geoapify.com/v1/geocode/reverse?lat=${latitude}&lon=${longitude}&format=json&apiKey=${
+        import.meta.env.VITE_GEOAPIFYKEY
+      }`,
+      {
+        method: "GET",
+        headers: {
+          "Content-Type": "Application/json",
+        },
+      }
+    ).then((location) =>
+      location.json().then((location) => {
+        setPlaceId(location.results[0].place_id);
+        setIsLoading(false);
+      })
+    );
+  }
+
+  // Find coordinates that are the closest to the middlepoint on land
+  async function findCoordinates(coordinates, middleLatitude, middleLongitude) {
+    let mini = 10000; // Default value so the first index can change
+    let distance;
+    let index;
+    coordinates.map((coordinates, indexe) => {
+      distance = getDistance(
+        { latitude: middleLatitude, longitude: middleLongitude },
+        { latitude: coordinates[1], longitude: coordinates[0] }
+      );
+      // Convert meters to miles
+      distance *= 0.000621371;
```

Comment:
let's store this 0.000621371 to a constant.

Reply:
Sounds good. Will do it this week after I finish polishing the UI